doom/config.org
#+title: Delta's Doom Emacs Config
#+author: Deluwuta (<https://github.com/deluwuta>)
#+description: Delta's personal Doom config.
#+startup: showeverything

* TABLE OF CONTENTS :toc:
- [[#dashboard][DASHBOARD]]
  - [[#custom-banner][Custom banner]]
- [[#fonts][FONTS]]
- [[#graphical-interface][GRAPHICAL INTERFACE]]
  - [[#line-configurations][Line configurations]]
- [[#mouse-support][MOUSE SUPPORT]]
- [[#org-mode][ORG MODE]]
  - [[#org-directory][Org directory]]
  - [[#enabling-org-bullets][Enabling Org Bullets]]
  - [[#headings-at-point][Headings at point]]
  - [[#disable-company][Disable Company]]
- [[#quality-of-life][QUALITY OF LIFE]]
- [[#rainbow-mode][RAINBOW MODE]]
- [[#scrolling][SCROLLING]]
- [[#theme][THEME]]

* DASHBOARD
** Custom banner

#+begin_src emacs-lisp

; If you want to use an image, change the path of the line below
(setq fancy-splash-image "~/Pictures/emacs_logo1_res.png")


; Ascii ART
(defun my-weebery-is-always-greater ()
  (let* ((banner '("⣇⣿⠘⣿⣿⣿⡿⡿⣟⣟⢟⢟⢝⠵⡝⣿⡿⢂⣼⣿⣷⣌⠩⡫⡻⣝⠹⢿⣿⣷"
        "⡆⣿⣆⠱⣝⡵⣝⢅⠙⣿⢕⢕⢕⢕⢝⣥⢒⠅⣿⣿⣿⡿⣳⣌⠪⡪⣡⢑⢝⣇"
        "⡆⣿⣿⣦⠹⣳⣳⣕⢅⠈⢗⢕⢕⢕⢕⢕⢈⢆⠟⠋⠉⠁⠉⠉⠁⠈⠼⢐⢕⢽"
        "⡗⢰⣶⣶⣦⣝⢝⢕⢕⠅⡆⢕⢕⢕⢕⢕⣴⠏⣠⡶⠛⡉⡉⡛⢶⣦⡀⠐⣕⢕"
        "⡝⡄⢻⢟⣿⣿⣷⣕⣕⣅⣿⣔⣕⣵⣵⣿⣿⢠⣿⢠⣮⡈⣌⠨⠅⠹⣷⡀⢱⢕"
        "⡝⡵⠟⠈⢀⣀⣀⡀⠉⢿⣿⣿⣿⣿⣿⣿⣿⣼⣿⢈⡋⠴⢿⡟⣡⡇⣿⡇⡀⢕"
        "⡝⠁⣠⣾⠟⡉⡉⡉⠻⣦⣻⣿⣿⣿⣿⣿⣿⣿⣿⣧⠸⣿⣦⣥⣿⡇⡿⣰⢗⢄"
        "⠁⢰⣿⡏⣴⣌⠈⣌⠡⠈⢻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣬⣉⣉⣁⣄⢖⢕⢕⢕"
        "⡀⢻⣿⡇⢙⠁⠴⢿⡟⣡⡆⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⣵⣵⣿"
        "⡻⣄⣻⣿⣌⠘⢿⣷⣥⣿⠇⣿⣿⣿⣿⣿⣿⠛⠻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿"
        "⣷⢄⠻⣿⣟⠿⠦⠍⠉⣡⣾⣿⣿⣿⣿⣿⣿⢸⣿⣦⠙⣿⣿⣿⣿⣿⣿⣿⣿⠟"
        "⡕⡑⣑⣈⣻⢗⢟⢞⢝⣻⣿⣿⣿⣿⣿⣿⣿⠸⣿⠿⠃⣿⣿⣿⣿⣿⣿⡿⠁⣠"
        "⡝⡵⡈⢟⢕⢕⢕⢕⣵⣿⣿⣿⣿⣿⣿⣿⣿⣿⣶⣶⣿⣿⣿⣿⣿⠿⠋⣀⣈⠙"
        "⡝⡵⡕⡀⠑⠳⠿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠿⠛⢉⡠⡲⡫⡪⡪⡣" ))
         (longest-line (apply #'max (mapcar #'length banner))))
    (put-text-property
     (point)
     (dolist (line banner (point))
       (insert (+doom-dashboard--center
                +doom-dashboard--width
                (concat line (make-string (max 0 (- longest-line (length line))) 32)))
               "\n"))
     'face 'doom-dashboard-banner)))

;; (setq +doom-dashboard-ascii-banner-fn #'my-weebery-is-always-greater)
(setq dashboard-center-content t)
#+end_src

* FONTS
Defining fonts (*pretends to be shocked*).
Settings related to fonts within Doom Emacs:
+ `doom-font' -- the primary font to use
+ `doom-variable-pitch-font' -- a non-monospace font (where applicable)
+ `doom-big-font' -- used for `doom-big-font-mode'; use this for presentations or streaming.
+ `doom-unicode-font' -- for unicode glyphs
+ `doom-serif-font' -- for the `fixed-pitch-serif' face

#+begin_src emacs-lisp
(setq doom-font
      (font-spec :family "FiraCode Nerd Font"
      ;; (font-spec :family "Hack Nerd Font"
                 :size 16
                 :weight 'medium)
      doom-variable-pitch-font
      (font-spec :family "FantasqueSansM Nerd Font"
                 :size 16
                 :weight 'medium))
(after! doom-themes
  (setq doom-themes-enable-bold t
        doom-themes-enable-italic t))

(custom-set-faces!
  '(font-lock-comment-face :slant italic)
  '(font-lock-keyword-face :weight bold))

(setq-default line-spacing 0.12)
#+end_src

* GRAPHICAL INTERFACE
La verdad es que tengo problemas mentales.

** Line configurations
#+begin_src emacs-lisp

; Some colors: #89dceb #f5c2e7 #f38ba8

(map! :leader
      :desc "Comment or uncomment lines" "l" #'comment-line)

(setq evil-normal-state-cursor '(box "#89dceb")
      evil-insert-state-cursor '(hbar "#89dceb")
      evil-visual-state-cursor '(hollow "#f38ba8"))

(setq display-line-numbers-type 'relative)
(setq display-line-numbers-mode 1)
(global-visual-line-mode t)

(custom-set-faces!
  '(line-number-current-line :foreground "#f5c2e7"
                             :slant italic
                             :weight bold))
#+end_src

* MOUSE SUPPORT
For the terminal version of Emacs.

#+begin_src emacs-lisp
(xterm-mouse-mode 1)
#+end_src

* ORG MODE
** Org directory
If you use `org' and don't want your org files in the default location below, change `org-directory'. It must be set before org loads!

#+begin_src emacs-lisp
(setq org-directory "~/org/")
#+end_src

** Enabling Org Bullets
Org-bullets gives us attrative bullets (ojalá en mi pecho) rather than asterisks

#+begin_src emacs-lisp
(add-hook 'org-mode-hook 'org-indent-mode)
(require 'org-bullets)
(add-hook 'org-mode-hook (lambda () (org-bullets-mode 1)))
#+end_src

** Headings at point
Insert Org headings at point, not after the current subtree (f*ck u Doom).
#+begin_src emacs-lisp
(after! org
  (setq org-insert-heading-respect-content nil))
#+end_src

** Disable Company
Is really anoying in Org.
#+begin_src emacs-lisp
(defun delta/adjust-org-company-backends ()
  (remove-hook 'after-change-major-mode-hook '+company-init-backends-h)
  (setq-local company-backends nil))

(add-hook! org-mode (delta/adjust-org-company-backends))
#+end_src

* QUALITY OF LIFE
Just a bunch of changes to try to reduce my sui**dal thoughts.

#+begin_src emacs-lisp
(setq confirm-kill-emacs nil) ; No more asking If I want to quit
(setq company-minimum-prefix-length 1)

; Why did it split ON TOP???
(after! evil
  (setq evil-vsplit-window-right t
        evil-split-window-below t))

(setq +lsp-company-backends '(:separate company-yasnippet company-capf))
#+end_src

* RAINBOW MODE
Idk why do I have to *explicitly* set this (Thank you DT).

#+begin_src emacs-lisp
(define-globalized-minor-mode global-rainbow-mode rainbow-mode
  (lambda ()
    (when (not (memq major-mode
                     (list 'org-agenda-mode)))
      (rainbow-mode 1))))
(global-rainbow-mode 1)
#+end_src

* SCROLLING
By default Doom Emacs fixes the "jumpy" scroll behaviour that Vanilla Emacs have, but I'll tweak it a bit.

#+begin_src emacs-lisp
(setq redisplay-dont-pause t
  scroll-margin 10
  scroll-step 1
  scroll-conservatively 10000
  scroll-perserve-screen-position 1
  mouse-wheel-progressive-speed t ;; (don't) accelerate scrolling
  ;mouse-whell-scroll-amount '(10)
  mouse-whell-follow-mouse 't
)
#+end_src

* THEME
Doom has a *lot* of themes but Catppuccin remains superior. To use Catppuccin as a theme is needed to add it on the package.el file first.

#+begin_src emacs-lisp
;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:

(setq doom-theme 'catppuccin)
(setq catppuccin-flavor 'macchiato) ; 'latte / 'macchiato / 'latte / 'mocha

#+end_src


.dwm/oldconfigdefh.txt
/* See LICENSE file for copyright and license details. */

/* appearance */
static const unsigned int borderpx  = 3;        /* border pixel of windows */
static const unsigned int gappx     = 8;        /* gaps between windows */
static const unsigned int snap      = 32;       /* snap pixel */
static const int showbar            = 1;        /* 0 means no bar */
static const int topbar             = 1;        /* 0 means bottom bar */
static const char *fonts[]          = { "Hack:size=11" };
static const char dmenufont[]       = "Hack:size=11";
static const char col_gray1[]       = "#222222";
static const char col_gray2[]       = "#444444";
static const char col_gray3[]       = "#bbbbbb";
static const char col_gray4[]       = "#eeeeee";
static const char col_cyan[]        = "#005577";
static const char *colors[][3]      = {
	/*               fg         bg         border   */
	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
};

/* tagging */
static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };

static const Rule rules[] = {
	/* xprop(1):
	 *	WM_CLASS(STRING) = instance, class
	 *	WM_NAME(STRING) = title
	 */
	/* class      instance    title       tags mask     isfloating   monitor */
	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
};

/* layout(s) */
static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
static const int nmaster     = 1;    /* number of clients in master area */
static const int resizehints = 0;    /* 1 means respect size hints in tiled resizals */
static const int lockfullscreen = 1; /* 1 will force focus on the fullscreen window */

static const Layout layouts[] = {
	/* symbol     arrange function */
	{ "[]=",      tile },    /* first entry is default */
	{ "><>",      NULL },    /* no layout function means floating behavior */
	{ "[M]",      monocle },
	{ "|M|",      centeredmaster },
	{ ">M>",      centeredfloatingmaster },
    { NULL,       NULL },
};

/* key definitions */
#define MODKEY Mod4Mask

#define TAGKEYS(KEY,TAG)												\
	{1, {{MODKEY, KEY}},								view,           {.ui = 1 << TAG} },	\
	{1, {{MODKEY|ControlMask, KEY}},					toggleview,     {.ui = 1 << TAG} }, \
	{1, {{MODKEY|ShiftMask, KEY}},						tag,            {.ui = 1 << TAG} }, \
	{1, {{MODKEY|ControlMask|ShiftMask, KEY}},			toggletag,      {.ui = 1 << TAG} },

/* helper for spawning shell commands in the pre dwm-5.0 fashion */
#define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }

/* commands */
static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
static const char *termcmd[]  = { "st", NULL };
static const char *emacs[]  = { "emacs", NULL };

static Keychord keychords[] = {
	/* Keys        function        argument */
	{1, {{MODKEY, XK_p}},							spawn,          {.v = dmenucmd } },
	{2, {{MODKEY, XK_e}, {MODKEY, XK_e}},			spawn,          {.v = termcmd } },

    /* My own keybindings */
    {1, {{MODKEY, XK_Return}},      spawn,       SHCMD("alacritty -e fish")},
    {1, {{MODKEY, XK_e}}, spawn, SHCMD("emacsclient -c -a 'emacs'")},
 
	{1, {{MODKEY, XK_b}},							togglebar,      {0} },
	{1, {{MODKEY, XK_j}},							focusstack,     {.i = +1 } },
	{1, {{MODKEY, XK_k}},							focusstack,     {.i = -1 } },
	{1, {{MODKEY, XK_i}},							incnmaster,     {.i = +1 } },
	{1, {{MODKEY, XK_d}},							incnmaster,     {.i = -1 } },
	{1, {{MODKEY, XK_h}},							setmfact,       {.f = -0.05} },
	{1, {{MODKEY, XK_l}},							setmfact,       {.f = +0.05} },
	{1, {{MODKEY|ShiftMask, XK_Return}},			zoom,           {0} },
	{1, {{MODKEY, XK_Tab}},							view,           {0} },
	{1, {{MODKEY|ShiftMask, XK_c}},					killclient,     {0} },
	{1, {{MODKEY, XK_t}},							setlayout,      {.v = &layouts[0]} },
	{1, {{MODKEY, XK_f}},							setlayout,      {.v = &layouts[1]} },
	{1, {{MODKEY, XK_m}},							setlayout,      {.v = &layouts[2]} },
    {1, {{MODKEY|ControlMask, XK_comma}},           cyclelayout,    {.i = -1} },
    {1, {{MODKEY|ControlMask, XK_period}},          cyclelayout,    {.i = +1} },
    {1, {{MODKEY, XK_u}},                           setlayout,      {.v = &layouts[3]} },
    {1, {{MODKEY, XK_o}},                           setlayout,      {.v = &layouts[4]} },
	{1, {{MODKEY, XK_space}},						setlayout,      {0} },
	{1, {{MODKEY|ShiftMask, XK_space}},				togglefloating, {0} },
	{1, {{MODKEY, XK_0}},							view,           {.ui = ~0 } },
	{1, {{MODKEY|ShiftMask, XK_0}},					tag,            {.ui = ~0 } },
	{1, {{MODKEY, XK_comma}},						focusmon,       {.i = -1 } },
	{1, {{MODKEY, XK_period}},						focusmon,       {.i = +1 } },
	{1, {{MODKEY|ShiftMask, XK_comma}},				tagmon,         {.i = -1 } },
	{1, {{MODKEY|ShiftMask, XK_period}},			tagmon,         {.i = +1 } },

	TAGKEYS(                        XK_1,                      0)
	TAGKEYS(                        XK_2,                      1)
	TAGKEYS(                        XK_3,                      2)
	TAGKEYS(                        XK_4,                      3)
	TAGKEYS(                        XK_5,                      4)
	TAGKEYS(                        XK_6,                      5)
	TAGKEYS(                        XK_7,                      6)
	TAGKEYS(                        XK_8,                      7)
	TAGKEYS(                        XK_9,                      8)

	{1, {{MODKEY|ShiftMask, XK_q}},      quit,           {0} },
	{1, {{MODKEY|ShiftMask, XK_r}},      quit,           {1} }, 
};

/* button definitions */
/* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
static Button buttons[] = {
	/* click                event mask      button          function        argument */
	{ ClkLtSymbol,          0,              Button1,        setlayout,      {0} },
	{ ClkLtSymbol,          0,              Button3,        setlayout,      {.v = &layouts[2]} },
	{ ClkWinTitle,          0,              Button2,        zoom,           {0} },
	{ ClkStatusText,        0,              Button2,        spawn,          {.v = termcmd } },
	{ ClkClientWin,         MODKEY,         Button1,        movemouse,      {0} },
	{ ClkClientWin,         MODKEY,         Button2,        togglefloating, {0} },
	{ ClkClientWin,         MODKEY,         Button3,        resizemouse,    {0} },
	{ ClkTagBar,            0,              Button1,        view,           {0} },
	{ ClkTagBar,            0,              Button3,        toggleview,     {0} },
	{ ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
	{ ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
};



dwmProject/config.h
/*
                         
    /\                   
   /  \   __   __  _   _ 
  / /\ \ / / _ \ \| | | |
 / /__\ \ |_/ \_| | |_| |
/________\___^___/| ._,_|
                  | |    
                  |_|    

Delta's Δωμ config file. 
*/

#define STATUSBAR "dwmblocks"

// Theme
#include "theme.h"

// Autostarting
static const char *const autostart[] = {
    "$HOME/.config/dwm/autostart.sh &",
	NULL /* terminate */
};

/* Systray */
static const unsigned int systraypinning = 0; /* 0: sloppy systray follows selected monitor, >0: pin systray to monitor X */
static const unsigned int systrayonleft  = 0; /* 0: systray in the right corner, >0: systray on left of status text */
static const unsigned int systrayspacing = 6; /* systray spacing */
static const int systraypinningfailfirst = 1; /* 1: if pinning fails, display systray on the first monitor, False: display systray on the last monitor*/
static const int showsystray = 1; /* 0 means no systray */

/* appearance */
static const unsigned int borderpx = 3;  /* border pixel of windows */
static const unsigned int gappx    = 6; /* Gaps between windows */
static const unsigned int snap     = 32; /* snap pixel */

static const int showbar = 1;  /* 0 means no bar */
static const int topbar  = 1;  /* 0 means bottom bar */

static const char *fonts[] = { "Hack:size=14" };
static const char dmenufont[] = "FiraCode Nerd Font:size=11";

// #faedff #eeeeee //#454545
static const char *colors[][3]      = {
	/*               fg         bg         border   */
	[SchemeNorm]      = { pure_white, darkest_dark, darkest_dark },
	[SchemeSel]       = { darkest_dark, white2, purple },
    [SchemeUrg]      = { pink, darker_black, darker_black },
	[SchemeTitleNorm] = { light_grey, darker_black, grey},
	[SchemeTitleSel]  = { pink, pink, pink},
	[SchemeTagsNorm]  = { pink, pink, pink},
	[SchemeTagsSel]   = { pink, pink, pink},
};

/* tagging */
static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };


/* Tag colors stuff */
/*static const int tagschemes[] = { SchemeTag1, SchemeTag2, SchemeTag3,*/
/*                                  SchemeTag4, SchemeTag5, SchemeTag6,*/
/*                                  SchemeTag7, SchemeTag8, SchemeTag9 };*/
/*static const int titleschemes[] = { SchemeTitle1, SchemeTitle2, SchemeTitle3,*/
/*                                    SchemeTitle4, SchemeTitle5, SchemeTitle6,*/
/*                                    SchemeTitle7, SchemeTitle8, SchemeTitle9 };*/

static const Rule rules[] = {
	/* xprop(1):
	 *	WM_CLASS(STRING) = instance, class
	 *	WM_NAME(STRING) = title
	 */
	/* class      instance    title       tags mask     isfloating   monitor */
	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
};

/* layout(s) */
static const float mfact        = 0.50; /* factor of master area size [0.05..0.95] */
static const int nmaster        = 1;    /* number of clients in master area */
static const int resizehints    = 1;    /* 1 means respect size hints in tiled resizals */
static const int lockfullscreen = 1;    /* 1 will force focus on the fullscreen window */

static const Layout layouts[] = {
	/* symbol     arrange function */
	{ "[]=",      tile },    /* first entry is default */
	{ "><>",      NULL },    /* no layout function means floating behavior */
	{ "[M]",      monocle },
};

/* key definitions */
#define MODKEY Mod4Mask
#define ALTKEY Mod1Mask

#define TAGKEYS(KEY,TAG) \
	{ MODKEY,                       KEY, view,       {.ui = 1 << TAG} }, \
	{ MODKEY|ControlMask,           KEY, toggleview, {.ui = 1 << TAG} }, \
	{ MODKEY|ShiftMask,             KEY, tag,        {.ui = 1 << TAG} }, \
	{ MODKEY|ControlMask|ShiftMask, KEY, toggletag,  {.ui = 1 << TAG} },

/* helper for spawning shell commands in the pre dwm-5.0 fashion */
#define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }

/* commands */

// Dmenu launcher lol
static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
static const char *dmenucmd[] = { 
	"dmenu_run", 
	"-m", dmenumon, 
	"-fn", dmenufont, 
	"-nb", darkest_dark, 
	"-nf", pure_white, 
	"-sb", white2, 
	"-sf", darkest_dark, 
	NULL };

static const char launcher[] = {
    "rofi -show"
};

static const char *termcmd[]  = { "wezterm", NULL };

static const Key keys[] = {
	/* modifier                     key        function        argument */
	// Yeee
	/* { MODKEY, XK_p	   , spawn, {.v = launcher } }, */
	{ MODKEY, XK_p	   , spawn, SHCMD(launcher) },
	{ MODKEY, XK_Return, spawn, {.v = termcmd } },

	// Quiting and restarting
	{ MODKEY|ControlMask, XK_q, quit, {0} },
	{ MODKEY|ControlMask, XK_r, quit, {1} },

	// Killin
	{ MODKEY|ShiftMask, XK_c, killclient, {0} },

	// Basic stuff
	/* { MODKEY, XK_b, spawn, SHCMD("brave --password-store=kwallet5")}, */
	{ MODKEY, XK_b, spawn, SHCMD("thorium-browser")},
    { MODKEY|ShiftMask, XK_s, spawn, SHCMD("flameshot gui")},

  // Multimedia
    {0, XF86XK_AudioRaiseVolume, spawn, SHCMD("amixer -D pulse set Master 5%+ unmute > /dev/null; pkill -RTMIN+6 dwmblocks")},
    {0, XF86XK_AudioLowerVolume, spawn, SHCMD("amixer -D pulse set Master 5%- unmute > /dev/null; pkill -RTMIN+6 dwmblocks")},

    {0, XF86XK_MonBrightnessUp, spawn, SHCMD("brightnessctl s 5%+ ; pkill -RTMIN+8 dwmblocks")},
	{0, XF86XK_MonBrightnessDown, spawn, SHCMD("brightnessctl s 5%- ; pkill -RTMIN+8 dwmblocks")},

	// Force to tile | Hide bar
	{ MODKEY|ShiftMask  , XK_t, togglefloating, {0} },
	{ MODKEY|ControlMask, XK_h, togglebar, {0} },

	// Windows swapin
	{ MODKEY|ShiftMask, XK_space, zoom, {0} },

	// Window movement and size manip
	{ MODKEY, XK_j, focusstack, {.i = +1 } },
	{ MODKEY, XK_k, focusstack, {.i = -1 } },
	{ MODKEY, XK_h, setmfact, {.f = -0.05} },
	{ MODKEY, XK_l, setmfact, {.f = +0.05} },

	{ MODKEY, XK_Tab  , view, {0} },

	/* { MODKEY, XK_t    , setlayout, {.v = &layouts[0]} }, */
	/* { MODKEY, XK_f    , setlayout, {.v = &layouts[1]} }, */
	/* { MODKEY, XK_m    , setlayout, {.v = &layouts[2]} }, */

    // Reset layout
	{ MODKEY, XK_Escape, setlayout, {0} },

	{ MODKEY, XK_0, view, {.ui = ~0 } },
	{ MODKEY|ShiftMask, XK_0, tag, {.ui = ~0 } },


	//{ MODKEY, XK_i, incnmaster, {.i = +1 } },
	//{ MODKEY, XK_d, incnmaster, {.i = -1 } },

	// Multimonitor keys
	{ MODKEY, XK_comma,  focusmon, {.i = -1 } },
	{ MODKEY, XK_period, focusmon, {.i = +1 } },
	{ MODKEY|ShiftMask, XK_comma, tagmon, {.i = -1 } },
	{ MODKEY|ShiftMask, XK_period, tagmon, {.i = +1 } },

	TAGKEYS(XK_1, 0)
	TAGKEYS(XK_2, 1)
	TAGKEYS(XK_3, 2)
	TAGKEYS(XK_4, 3)
	TAGKEYS(XK_5, 4)
	TAGKEYS(XK_6, 5)
	TAGKEYS(XK_7, 6)
	TAGKEYS(XK_8, 7)
	TAGKEYS(XK_9, 8)
};

/* button definitions */
/* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
static const Button buttons[] = {
	/* click                event mask      button          function        argument */
	{ ClkLtSymbol,          0,              Button1,        setlayout,      {0} },
	{ ClkLtSymbol,          0,              Button3,        setlayout,      {.v = &layouts[2]} },
	{ ClkWinTitle,          0,              Button2,        zoom,           {0} },
	{ ClkStatusText,        0,              Button2,        spawn,          {.v = termcmd } },
	{ ClkClientWin,         MODKEY,         Button1,        movemouse,      {0} },
	{ ClkClientWin,         MODKEY,         Button2,        togglefloating, {0} },
	{ ClkClientWin,         MODKEY,         Button3,        resizemouse,    {0} },
	{ ClkTagBar,            0,              Button1,        view,           {0} },
	{ ClkTagBar,            0,              Button3,        toggleview,     {0} },
	{ ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
	{ ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
};

